processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByRowProcessor.java

```diff
@@ -0,0 +1,508 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.groupby.epinephelinae;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.metamx.common.ISE;
+import com.metamx.common.guava.Accumulator;
+import com.metamx.common.guava.BaseSequence;
+import com.metamx.common.guava.CloseQuietly;
+import com.metamx.common.guava.FilteredSequence;
+import com.metamx.common.guava.ResourceClosingSequence;
+import com.metamx.common.guava.Sequence;
+import com.metamx.common.logger.Logger;
+import io.druid.collections.BlockingPool;
+import io.druid.collections.ResourceHolder;
+import io.druid.data.input.MapBasedRow;
+import io.druid.data.input.Row;
+import io.druid.granularity.AllGranularity;
+import io.druid.granularity.QueryGranularity;
+import io.druid.query.Query;
+import io.druid.query.QueryContextKeys;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.dimension.DimensionSpec;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.query.filter.DruidLongPredicate;
+import io.druid.query.filter.DruidPredicateFactory;
+import io.druid.query.filter.Filter;
+import io.druid.query.filter.ValueMatcher;
+import io.druid.query.filter.ValueMatcherFactory;
+import io.druid.query.groupby.GroupByQuery;
+import io.druid.query.groupby.GroupByQueryConfig;
+import io.druid.query.groupby.strategy.GroupByStrategyV2;
+import io.druid.query.groupby.epinephelinae.GroupByMergingQueryRunnerV2.GroupByMergingKey;
+import io.druid.query.groupby.epinephelinae.GroupByMergingQueryRunnerV2.GroupByMergingKeySerdeFactory;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.DimensionSelector;
+import io.druid.segment.FloatColumnSelector;
+import io.druid.segment.LongColumnSelector;
+import io.druid.segment.ObjectColumnSelector;
+import io.druid.segment.column.Column;
+import io.druid.segment.column.ColumnCapabilities;
+import io.druid.segment.data.IndexedInts;
+import io.druid.segment.filter.BooleanValueMatcher;
+import io.druid.segment.filter.Filters;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class GroupByRowProcessor
```

Comment:
This and GroupByMergingQueryRunnerV2 should be sharing code if at all possible. Is there any reason they can't?